python/lsst/afw/geom/testUtils.py

```diff
@@ -91,3 +100,759 @@ def __iter__(self):
+        - isBaseSkyFrame: a boolean
+        - isCurrSkyFrame: a boolean
+        - isBasePermuted: a boolean
+        - isCurrPermuted: a boolean
+        """
+        baseInd = frameSet.getBase()
+        currInd = frameSet.getCurrent()
+        isBaseSkyFrame = frameSet.getFrame(baseInd).getClass() == "SkyFrame"
+        isCurrSkyFrame = frameSet.getFrame(currInd).getClass() == "SkyFrame"
+        if not (isBaseSkyFrame or isCurrSkyFrame):
+            return
+
+        baseList = [False, True] if isBaseSkyFrame else [False]
+        currList = [False, True] if isCurrSkyFrame else [False]
+        for isBasePermuted in baseList:
+            for isCurrPermuted in currList:
+                frameSetCopy = frameSet.copy()
+                if isBasePermuted:
+                    assert isBaseSkyFrame
+                    frameSetCopy.setCurrent(baseInd)
+                    frameSetCopy.permAxes([2, 1])
+                    frameSetCopy.setCurrent(currInd)
+                if isCurrPermuted:
+                    assert isCurrSkyFrame
+                    frameSetCopy.permAxes([2, 1])
+                yield (frameSetCopy, isBaseSkyFrame, isCurrSkyFrame, isBasePermuted, isCurrPermuted)
+
+    @staticmethod
+    def makeJacobian(nIn, nOut, inArray):
+        """Make a Jacobian matrix for the equation described by makeTwoWayPolyMap.
+
+        Parameters
+        ----------
+        nIn, nOut : integers
+            the dimensions of the input and output data; see makeTwoWayPolyMap
+        inArray : ndarray
+            an array of size `nIn` representing the point at which the Jacobian
+            is measured
+
+        Returns
+        ----------
+        J : numpy.ndarray
+            an nOut x nIn array of first derivatives
+        """
+        baseCoeff = 2.0 * 0.001
+        coeffs = np.empty((nOut, nIn))
+        for iOut in range(nOut):
+            coeffOffset = baseCoeff * iOut
+            for iIn in range(nIn):
+                coeffs[iOut, iIn] = baseCoeff * (iIn + 1) + coeffOffset
+                coeffs[iOut, iIn] *= inArray[iIn]
+        assert coeffs.ndim == 2
+        assert coeffs.shape == (nOut, nIn)
+        # Avoid spurious errors when comparing to a simplified array
+        return coeffs
+
+    def checkTransformation(self, transform, mapping, msg=""):
+        """Check tranForward and tranInverse for a transform
+
+        Parameters
+        ----------
+        transform : Transform
+            The transform to check
+        mapping : astshim.Mapping
+            The mapping the transform should use. This mapping
+            must contain valid forward or inverse transformations,
+            but they need not match if both present. Hence the
+            mappings returned by make*PolyMap are acceptable.
+        msg : string
+            Error message suffix describing test parameters
+        """
+        fromEndpoint = transform.getFromEndpoint()
+        toEndpoint = transform.getToEndpoint()
```

Comment:
I'm not sure I understand the scope or intent of your remark. It is true that users will rarely get or manipulate an endpoint, but it seems appropriate for unit tests to use them to transform data from generic form to the specific form taken and returned by a Transform. It allows writing generic tests that can test many types of transforms without worrying about the specific data type of data that a given transform manipulates.